packages/kit/src/runtime/server/page/render.js

```diff
@@ -297,11 +297,19 @@ export async function render_response({
 
 		const blocks = [];
 
-		const properties = [
-			paths.assets && `assets: ${s(paths.assets)}`,
-			`base: ${base_expression}`,
-			`env: ${!client.uses_env_dynamic_public || state.prerendering ? null : s(public_env)}`
-		].filter(Boolean);
+		// when serving a prerendered page in an app that uses $env/dynamic/public, we must
+		// import the env.js module so that it evaluates before any user code can evaluate
```

Comment:
```suggestion
		// import the env.js module so that it evaluates before any user code can evaluate.
		// TODO revert to using top-level await once https://bugs.webkit.org/show_bug.cgi?id=242740 is fixed
		// https://github.com/sveltejs/kit/pull/11601
```